src/components/connect.js

```diff
@@ -17,34 +19,33 @@ function getDisplayName(WrappedComponent) {
 let nextVersion = 0
 
 function connectFactory(defaults = {}) {
-  function mergeHeaders(options) {
-    if (options.hasOwnProperty('headers')) {
-      checks.headers(options.headers)
-    }
-    return Object.assign({}, defaults.headers, options.headers)
-  }
-
   function connectImpl(map, options = {}) {
-    warning(Object.getOwnPropertyNames(options).length === 0,
-      'The options argument is deprecated in favor of `connect.defaults()` ' +
-      'calls. In a future release, support will be removed.'
-    )
-
-    options.headers = mergeHeaders(options)
-    const finalOptions = Object.assign({}, defaults, options)
+    let finalOptions = defaults
+    if ('withRef' in options) {
+      warning(false,
+        'The options argument is deprecated in favor of `connect.defaults()`.' +
+        ' In a future release, support will be removed.'
```

Comment:
IIRC, this warning is tested, so changing the wording may need a change in the test.